src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -384,12 +384,13 @@ private bool ConnectionNameMatchesCurrentApp(DurableClient client)
         async Task IDurableOrchestrationClient.TerminateAsync(string instanceId, string reason)
         {
             OrchestrationState state = await this.GetOrchestrationInstanceStateAsync(instanceId);
-            if (IsOrchestrationRunning(state))
+            if (IsOrchestrationRunning(state) || IsOrchestrationSuspended(state))
```

Comment:
I think we want to consider Suspended a "running" state. This check is meant to cover all states other than failed or completed.

Reply:
Same concerned pointed out by David in the same class: https://github.com/Azure/azure-functions-durable-extension/pull/2227#discussion_r921341433 This is what I answered there:

Aren't they different, though? For instance, while here it makes sense to check for running or suspended, in other instances it makes sense to check just one or the other (see SuspendAsync and TerminateAsync)

Reply:
I solved this by renaming to make clear when each helper method should be used and for what purpose:
- `IsOrchestrationAvailable()` is used by `TerminateAsync() `and `RaiseEventInternalAsync()`.
- `IsOrchestrationSuspendable()` is used by `SuspendAsync() ` only
- `IsOrchestrationSuspended()`  is used by `ResumeAsync()`.